editors/code/src/net.ts

```diff
@@ -116,7 +117,41 @@ export async function download(opts: DownloadOpts) {
         }
     );
 
+    // Try to rename a running server to avoid EPERM on Windows
+    // NB: this can lead to issues if a running Code instance tries to restart the server.
+    try {
+        await vscode.workspace.fs.rename(opts.dest, oldServerPath, { overwrite: true });
+        log.info(`Renamed old server binary ${opts.dest.fsPath} to ${oldServerPath.fsPath}`);
+    } catch (err) {
+        const fsErr = err as vscode.FileSystemError;
+        if (!fsErr || fsErr.code !== "FileNotFound") {
+            log.error(`Cannot rename running instance: ${err}`);
+        }
+    }
```

Comment:
What does `!fsErr` do here?

Reply:
I'm checking for null, in case it's a different exception.

Reply:
The reminder in [this section](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions) of the TS docs seems to suggest that `fsErr` can never be null.

Reply:
Oops, thanks. I think Error has a name field, maybe that's what it's supposed to be used for?